src/preprocess_info.cc

```diff
@@ -158,7 +195,236 @@ Napi::Value PreProcessInfo::GetMeanVariant(const Napi::CallbackInfo& info) {
+        .ThrowAsJavaScriptException();
+    return;
+  };
+
+  if (index >= numberOfChannels || index < 0) {
+    std::string errorLog =
+        "pre process index " + std::to_string(index) + " is out of bounds.";
+    Napi::Error::New(env, errorLog).ThrowAsJavaScriptException();
+    return;
+  };
+
+  ie::PreProcessChannel::Ptr channel = preInfo[index];
+  if (newChannel.Has("stdScale")) {
+    if (newChannel.Get("stdScale").IsNumber()) {
+      channel->stdScale = newChannel.Get("stdScale").ToNumber().FloatValue();
+    } else {
+      Napi::TypeError::New(env, "Wrong stdScale").ThrowAsJavaScriptException();
+    }
+  }
+
+  if (newChannel.Has("meanValue")) {
+    if (newChannel.Get("meanValue").IsNumber()) {
+      channel->meanValue = newChannel.Get("meanValue").ToNumber().FloatValue();
+    } else {
+      Napi::TypeError::New(env, "Wrong meanValue").ThrowAsJavaScriptException();
+    }
+  }
+
+  if (newChannel.Has("meanData")) {
+    if (!newChannel.Get("meanData").IsObject()) {
+      Napi::TypeError::New(env, "Wrong meanData").ThrowAsJavaScriptException();
+      return;
+    }
+
+    Napi::Object meanData = newChannel.Get("meanData").ToObject();
+
+    if (!meanData.Has("desc") || !meanData.Has("data")) {
+      Napi::TypeError::New(env, "Wrong meanData").ThrowAsJavaScriptException();
+      return;
+    }
+
+    if (!meanData.Get("desc").IsObject()) {
+      Napi::TypeError::New(env, "Wrong desc").ThrowAsJavaScriptException();
+      return;
+    }
+
+    if (!meanData.Get("data").IsArrayBuffer()) {
+      Napi::TypeError::New(env, "Wrong data").ThrowAsJavaScriptException();
+      return;
+    }
+
+    Napi::Object desc = meanData.Get("desc").ToObject();
+    if (!checkDesc(desc)) {
+      Napi::TypeError::New(env, "Wrong desc").ThrowAsJavaScriptException();
+      return;
+    }
+
+    Napi::ArrayBuffer data =
+        Napi::ArrayBuffer::ArrayBuffer(env, meanData.Get("data"));
+    std::string precision_string = desc.Get("precision").ToString();
+    std::string layout_string = desc.Get("layout").ToString();
+    Napi::ArrayBuffer dims =
+        Napi::ArrayBuffer::ArrayBuffer(env, desc.Get("dims"));
+
+    size_t datalength = data.ByteLength();
+    size_t dimslength = dims.ByteLength() / sizeof(uint32_t);
+    const uint32_t* dimArray = reinterpret_cast<uint32_t*>(dims.Data());
+    void* buffer = data.Data();
+
+    try {
+      ie::SizeVector dims_vector = {};
+      for (size_t i = 0; i < dimslength; i++) {
+        dims_vector.push_back(dimArray[i]);
+      }
+      ie::Precision precision = utils::GetPrecisionByName(precision_string);
+      ie::Layout layout = utils::GetLayoutByName(layout_string);
+      ie::TensorDesc tensor(precision, dims_vector, layout);
+      ie::Blob::Ptr meanDataBlob;
+      switch (precision) {
+        case ie::Precision::FP32:
+          meanDataBlob = ie::make_shared_blob<float>(
+              tensor, reinterpret_cast<float_t*>(buffer), datalength);
```

Comment:
Please don't create the `meanDataBlob` from the `buffer`, as `buffer` could be released in JavaScript garbage collector. I suggest to allocate the memory of `meanDataBlob` and copy the data from `buffer` to `meanDataBlob`'s memory.

Reply:
```
case ie::Precision::FP32:
       meanDataBlob = ie::make_shared_blob<float>(
            tensor, reinterpret_cast<float*>(malloc(datalength)), datalength);
       memcpy(ie::as<ie::MemoryBlob>(meanDataBlob)->wmap(), buffer, datalength);
       break;
```
I have done it by using above way. This implementation passes the tests. But I am not sure it is good enough.